File: app/src/main/java/cn/ifafu/ifafu/mvp/syllabus_setting/SyllabusSettingModel.java
package cn.ifafu.ifafu.mvp.syllabus_setting;

import android.content.Context;

import cn.ifafu.ifafu.data.entity.SyllabusSetting;
import cn.ifafu.ifafu.mvp.base.BaseZFModel;

public class SyllabusSettingModel extends BaseZFModel implements SyllabusSettingContract.Model {

    public SyllabusSettingModel(Context context) {
        super(context);
    }

    @Override
    public SyllabusSetting getSetting() {
        SyllabusSetting setting = repository.getSyllabusSetting();
        if (setting == null) {
            setting = new SyllabusSetting(repository.getLoginUser().getAccount());
            repository.saveSyllabusSetting(setting);
        }
        return setting;
    }

    @Override
    public void save(SyllabusSetting setting) {
        repository.saveSyllabusSetting(setting);
    }
}

File: app/src/main/java/cn/ifafu/ifafu/mvp/syllabus_setting/SyllabusSettingModel.kt
package cn.ifafu.ifafu.mvp.syllabus_setting

import android.content.Context
import cn.ifafu.ifafu.base.mvp.BaseModel
import cn.ifafu.ifafu.entity.SyllabusSetting

class SyllabusSettingModel(context: Context) : BaseModel(context), SyllabusSettingContract.Model {

    override fun getSetting(): SyllabusSetting {
        return mRepository.getSyllabusSetting()
    }

    override suspend fun save(setting: SyllabusSetting) {
        mRepository.saveSyllabusSetting(setting)
    }
}
